daemon/mgr/container.go

```diff
@@ -855,7 +863,126 @@ func (mgr *ContainerManager) Update(ctx context.Context, name string, config *ty
 
 // Upgrade upgrades a container with new image and args.
 func (mgr *ContainerManager) Upgrade(ctx context.Context, name string, config *types.ContainerUpgradeConfig) error {
-	// TODO
+	c, err := mgr.container(name)
+	if err != nil {
+		return err
+	}
+
+	c.Lock()
+	defer c.Unlock()
+
+	// check the image existed or not, and convert image id to image ref
+	image, err := mgr.ImageMgr.GetImage(ctx, config.Image)
+	if err != nil {
+		return errors.Wrap(err, "get image failed")
+	}
+
+	// FIXME: image.Name does not exist,so convert Repotags or RepoDigests to ref
+	ref := ""
+	if len(image.RepoTags) > 0 {
+		ref = image.RepoTags[0]
+	} else {
+		ref = image.RepoDigests[0]
```

Comment:
Is there possibility that this `image.RepoDigests` has a length of 0? 
also ping @YaoZengzeng @zeppp